data-prepper-plugins/mutate-event-processors/src/main/java/org/opensearch/dataprepper/plugins/processor/mutateevent/RenameKeyProcessor.java

```diff
@@ -59,14 +69,27 @@ public Collection<Record<Event>> doExecute(final Collection<Record<Event>> recor
                         continue;
                     }
 
-                    if (entry.getFromKey().equals(entry.getToKey()) || !recordEvent.containsKey(entry.getFromKey())) {
+                    if (Objects.nonNull(entry.getFromKey()) && (entry.getFromKey().equals(entry.getToKey()) || !recordEvent.containsKey(entry.getFromKey()))) {
                         continue;
                     }
-
                     if (!recordEvent.containsKey(entry.getToKey()) || entry.getOverwriteIfToKeyExists()) {
-                        final Object source = recordEvent.get(entry.getFromKey(), Object.class);
-                        recordEvent.put(entry.getToKey(), source);
-                        recordEvent.delete(entry.getFromKey());
+                        if(Objects.nonNull(entry.getFromKey())) {
+                            final Object source = recordEvent.get(entry.getFromKey(), Object.class);
+                            recordEvent.put(entry.getToKey(), source);
+                            recordEvent.delete(entry.getFromKey());
+                        }
+                        if(Objects.nonNull(entry.getFromKeyPattern())) {
+                            Map<String,Object> eventMap = recordEvent.toMap();
+                            Pattern pattern = Pattern.compile(entry.getFromKeyPattern());
```

Comment:
Please compile this in the constructor so that you can avoid recomputing this each time.

Reply:
Fixed it